src/routes/profiles.py

```
@@ -1,5 +1,65 @@
+                              jwt_manager: JWTAuthManagerInterface = Depends(get_jwt_auth_manager),
+                              s3_client: S3StorageInterface = Depends(get_s3_storage_client),
+                              db: AsyncSession = Depends(get_db),
+                              token: str = Depends(get_token)):
+    token_data = jwt_manager.decode_access_token(token)
+    expire_date = token_data.get["exp"]
+    if expire_date.replace(tzinfo=timezone.utc) < datetime.now(timezone.utc):
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="Token has expired.")
+    response = await db.execute(select(UserModel).join(UserProfileModel).where(UserModel.id == user_id))
+    user = response.scalar_one_or_none()
+    if not user or not user.is_active:
+        raise HTTPException(status_code=status.HTTP_401_UNAUTHORIZED, detail="User not found or not active.")
+    user_id_token = token_data.get("user_id")
+    response = await db.execute(select(UserModel).join(UserGroupModel).where(UserModel.id == user_id_token))
+    current_user = response.scalar_one_or_none()
+    if not current_user:
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail="Current user not found.")
+    if current_user.id != user_id and current_user.group.name != UserGroupEnum.ADMIN:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN,
+                            detail="You don't have permission to edit this profile.")
+    if user.profile:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail="User already has a profile.")
+    try:
+        content = await payload.avatar.read()
+        avatar_file = f"{user_id}_avatar.{payload.avatar.content_type.split("/")[1]}"
+        await s3_client.upload_file(
+            file_name=avatar_file,
+            file_type=content
+        )
+        avatar_url = await s3_client.get_file_url(avatar_file)
+    except Exception:
```

Catching a generic `Exception` is too broad and can hide unexpected errors unrelated to the file upload. While this meets the requirement to return a 500 error, it is better practice to catch a more specific exception if the `s3_client` provides one (e.g., `S3StorageError`).